### PR TITLE
Added a generic BSD detection to the Makefile

### PR DESCRIPTION
**Pull request by Mailaender**
A followup of https://github.com/OpenRA/OpenRA/pull/8405/ and https://github.com/OpenRA/OpenRA/pull/8083.

### CODE REVIEW

File: thirdparty/configure-native-deps.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+if [ -f Eluant.dll.config ]; then
+	exit 0
+fi
+
+for location in $locations ; do
+	for soname in $sonames ; do
+		if [ -f "$location/$soname" ]; then
+			liblua51=$soname
+			echo "Detected Lua 5.1 library at "$location/$soname
```
Comment (moviuro) [inline thirdparty/configure-native-deps.sh]: Tiny typo: ```"Detected Lua 5.1 library at $location/$soname"``` would be better and avoid word splitting

Comment (Mailaender) [inline thirdparty/configure-native-deps.sh]: Agreed.

**Comment by moviuro** [comment]
:+1: Looks good to me, and a lot of help, too!

**Comment by moviuro** [comment]
OK :+1: works great for me!

**Comment by obrakmann** [comment]
And still works on Linux :+1:

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/620340145a5bbc88c6df0849113be4a456f273f5%5E...620340145a5bbc88c6df0849113be4a456f273f5)

**Comment by Mailaender** [comment]
Updated the blurp at https://github.com/OpenRA/OpenRA

**Pull request merged**